### PR TITLE
Reject 'connections add' CLI request if URI provided is invalid

### LINKED ISSUE #12369: In 2.0.0b2/master, CLI "airflow connections add" is not handling invalid URIs properly
**Apache Airflow version**: 2.0.0b2

**Environment**: Docker

- **Cloud provider or hardware configuration**:
- **OS** (e.g. from /etc/os-release):
- **Kernel** (e.g. `uname -a`):
- **Install tools**:
- **Others**:

**What happened**:

In current 2.0.0b2, `airflow connections add` is not handling invalid URIs properly. 

For example, running CLI `airflow connections add --conn-uri xyz conn_dummy` will
- create the connection “conn_dummy” successfully (which should not be, IMO)
- in the connection created, it only has one attribute filled, which is “schema”, but the value added is “yz” (the value we provide is "xyz", i.e., the 1st element is removed. because of https://github.com/apache/airflow/blob/master/airflow/models/connection.py#L213)

airflow_0284e070228e___

**What you expected to happen**:

In my opinion, a validation should be done first to ensure the `conn-uri` provided is valid (at least have “scheme” and “netloc” available in the `urlparse` result), and reject if it's an invalid URI.

### PR DESCRIPTION
**Pull request by XD-DENG**
Closes https://github.com/apache/airflow/issues/12369

- Reject the request is the URI is invalid
- The validity is decided by availability of both 'scheme' and 'netloc' in the parse result

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/cli/commands/connection_command.py [hunk 1]
```diff
@@ -177,16 +185,19 @@ def connections_add(args):
...
+            msg = f'The URI provided to --conn-uri is invalid: {args.conn_uri}'
+            raise SystemExit(msg)
         for arg in alternative_conn_specs:
             if getattr(args, arg) is not None:
                 invalid_args.append(arg)
     elif not args.conn_type:
         missing_args.append('conn-uri or conn-type')
     if missing_args:
-        msg = 'The following args are required to add a connection:' + f' {missing_args!r}'
+        msg = f'The following args are required to add a connection: {missing_args!r}'
```
Comment (XD-DENG) [inline airflow/cli/commands/connection_command.py:197]: This line and line 200 are not really related to this PR's scope. But fixing them is too minor and maybe not worth creating separate PR for them. So fixing them together here.

**Review by turbaszek — APPROVED** [review]
